cmd/openfga/main_test.go

```diff
@@ -203,6 +204,210 @@ func newOpenFGATester(t *testing.T, args ...string) OpenFGATester {
+					ContextualTuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("timeslot:12_13", "user", "fga_user:anne"),
+					},
+					Expectation: true,
+				},
+			},
+		},
+		{
+			name: "cache_computed_userset_subproblem_with_contextual_tuple",
+			typeDefinitions: parser.MustParse(`
+			type user
+
+			type document
+			  relations
+			    define restricted: [user] as self
+			    define viewer: [user] as self but not restricted
+			`),
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+			},
+			assertions: []checktest.Assertion{
+				{
+					Tuple:            tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{},
+					Expectation:      true,
+				},
+				{
+					Tuple: tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("document:1", "restricted", "user:jon"),
+					},
+					Expectation: false,
+				},
+			},
+		},
+		{
+			name: "cached_direct_relationship_with_contextual_tuple",
+			typeDefinitions: parser.MustParse(`
+			type user
+
+			type document
+			  relations
+			    define viewer: [user] as self
+			`),
+			assertions: []checktest.Assertion{
+				{
+					Tuple:            tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{},
+					Expectation:      false,
+				},
+				{
+					Tuple: tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					},
+					Expectation: true,
+				},
+			},
+		},
+		{
+			name: "cached_direct_userset_relationship_with_contextual_tuple",
+			typeDefinitions: parser.MustParse(`
+			type user
+			
+			type group
+			  relations
+			    define restricted: [user] as self
+			    define member: [user] as self but not restricted
+
+			type document
+			  relations
+			    define viewer: [group#member] as self
+			`),
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+				tuple.NewTupleKey("group:eng", "member", "user:jon"),
+			},
+			assertions: []checktest.Assertion{
+				{
+					Tuple:            tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{},
+					Expectation:      true,
+				},
+				{
+					Tuple: tuple.NewTupleKey("document:1", "viewer", "user:jon"),
+					ContextualTuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("group:eng", "restricted", "user:jon"),
+					},
+					Expectation: false,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			createResp, err := client.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+				Name: test.name,
+			})
+			require.NoError(t, err)
+			require.NotPanics(t, func() { ulid.MustParse(createResp.GetId()) })
```

Comment:
🤔

Reply:
It's not, but why not assert expectations of the API? If it were to ever fail that'd be good to know anyways.